src/main/java/sx/blah/discord/api/internal/DiscordWS.java

```diff
@@ -227,10 +235,11 @@ public void onWebSocketError(Throwable cause) {
 
 	@Override
 	public void onWebSocketBinary(byte[] payload, int offset, int len) {
-		BufferedReader reader = new BufferedReader(new InputStreamReader(new InflaterInputStream(new ByteArrayInputStream(payload, offset, len))));
-		onWebSocketText(reader.lines().collect(Collectors.joining()));
-		try {
-			reader.close();
+		ByteArrayOutputStream out = new ByteArrayOutputStream(payload.length * 2);
```

Comment:
This should be in its own try-with. If write or onWebSocketText throws an exception this won't close. Yes, closing a ByteArrayOutputStream has no real effect, but it's best practice in my opinion to always close Closeable resources.